File: packages/rook_sdk_samsung_health/android/src/main/kotlin/io/tryrook/rook_sdk_samsung_health/mapper/ComplexMappers.kt
package io.tryrook.rook_sdk_samsung_health.mapper

import io.tryrook.rook_sdk_samsung_health.proto.BloodPressureObjectProto
import io.tryrook.rook_sdk_samsung_health.proto.PositionObjectProto
import io.tryrook.rook_sdk_samsung_health.proto.TemperatureObjectProto
import io.tryrook.rook_sdk_samsung_health.proto.VelocityObjectProto
import io.tryrook.sdk.samsung.domain.model.complex.BloodPressureObject
import io.tryrook.sdk.samsung.domain.model.complex.PositionObject
import io.tryrook.sdk.samsung.domain.model.complex.TemperatureObject
import io.tryrook.sdk.samsung.domain.model.complex.VelocityObject

fun BloodPressureObject.toObjectProto(): BloodPressureObjectProto {
    return BloodPressureObjectProto.newBuilder()
        .setSystolicMmHg(systolicMmHg)
        .setDiastolicMmHg(diastolicMmHg)
        .build()
}

fun PositionObject.toObjectProto(): PositionObjectProto {
    return PositionObjectProto.newBuilder()
        .setLatDeg(latDeg)
        .setLngDeg(lngDeg)
        .build()
}

fun TemperatureObject.toObjectProto(): TemperatureObjectProto {
    return TemperatureObjectProto.newBuilder()
        .setTemperatureCelsius(temperatureCelsius)
        .setMeasurementType(measurementType)
        .build()
}

fun VelocityObject.toObjectProto(): VelocityObjectProto {
    return VelocityObjectProto.newBuilder()
        .setSpeedMetersPerSecond(speedMetersPerSecond)
        .setDirection(direction)
        .build()
}

File: packages/rook_sdk_samsung_health/lib/src/data/mapper/complex_mappers.dart
import 'package:rook_sdk_core/rook_sdk_core.dart';
import 'package:rook_sdk_samsung_health/src/data/proto/protos.pb.dart';

extension BloodPressureObjectMapper on BloodPressureObjectProto {
  BloodPressureObject toDomain() {
    return BloodPressureObject(
      systolicMmHg: systolicMmHg,
      diastolicMmHg: diastolicMmHg,
    );
  }
}

extension PositionObjectMapper on PositionObjectProto {
  PositionObject toDomain() {
    return PositionObject(latDeg: latDeg, lngDeg: lngDeg);
  }
}

extension TemperatureObjectMapper on TemperatureObjectProto {
  TemperatureObject toDomain() {
    return TemperatureObject(
      temperatureCelsius: temperatureCelsius,
      measurementType: measurementType,
    );
  }
}

extension VelocityObjectMapper on VelocityObjectProto {
  VelocityObject toDomain() {
    return VelocityObject(
      speedMetersPerSecond: speedMetersPerSecond,
      direction: direction,
    );
  }
}
